source/common/http/conn_manager_utility.cc

```diff
@@ -131,6 +131,70 @@ void ConnectionManagerUtility::mutateRequestHeaders(Http::HeaderMap& request_hea
   if (config.tracingConfig()) {
     Tracing::HttpTracerUtility::mutateHeaders(request_headers, runtime);
   }
+  mutateXfccRequestHeader(request_headers, connection, config);
+}
+
+void ConnectionManagerUtility::mutateXfccRequestHeader(Http::HeaderMap& request_headers,
+                                                       Network::Connection& connection,
+                                                       ConnectionManagerConfig& config) {
+  if (config.forwardClientCert() != Http::ForwardClientCertType::AlwaysForwardOnly &&
+      !(connection.ssl() && connection.ssl()->peerCertificatePresented())) {
+    request_headers.removeForwardedClientCert();
+    return;
+  }
+  std::string clientCertDetails;
```

Comment:
Since `clientCertDetails` is only used when `SanitizeSet` or `AppendForward` set, process other cases before building it. Also, append directly to header->value() might be more efficient.